scipy/stats/_hypotests.py

```diff
@@ -956,6 +958,204 @@ def barnard_exact(table, alternative="two-sided", pooled=True, n=32):
+    Boschloo exact test, we can assert three different null hypotheses :
+
+    - :math:`H_0 : p_1 \geq p_2` versus :math:`H_1 : p_1 < p_2`,
+      with `alternative` = "less"
+
+    - :math:`H_0 : p_1 \leq p_2` versus :math:`H_1 : p_1 > p_2`,
+      with `alternative` = "greater"
+
+    - :math:`H_0 : p_1 = p_2` versus :math:`H_1 : p_1 \neq p_2`,
+      with `alternative` = "two-sided" (default one)
+
+    In order to compute Boschloo's exact test, we are using the Fisher's
+    pvalue as a statistic.
+
+    Boschloo's and Barnard's are both more powerful than Fisher's exact
+    test. From some local testing, it appears that Boschloo is slightly
+    more powerfull than Barnard's procedure, but a lot slower.
+
+    .. versionadded:: 1.7.0
+
+    References
+    ----------
+    .. [1] R.D. Boschloo. "Raised conditional level of significance for the
+       2 x 2-table when testing the equality of two probabilities", 
+       STATISTICA NEERLANDICA, 24(1), 1970
+
+    .. [2] "Boschloo's test", Wikipedia,
+       https://en.wikipedia.org/wiki/Boschloo%27s_test
+
+    .. [3] Lise M. Saari et al. "Employee attitudes and job satisfaction",
+       Human Resource Management, 43(4), 395-407, 2004,
+       :doi:`10.1002/hrm.20032`.
+
+    Examples
+    --------
+    In the following example, we consider the article "Employee
+    attitudes and job satisfaction" [3]_
+    which reports the results of a survey from 63 Scientists and 117 College
+    Professors. Of the 63 Scientists, 31 said they were very satisfied with
+    their jobs, whereas 74 of the College Professors were very satisfied
+    with their work. Are there strong evidence suggesting that College
+    Professors are more happy with their works than Scientists?
+    The following table summurize the data mentionned above::
+
+                         College Professors   Scientists
+        Very Satisfied   74                     31
+        Dissatisfied     43                     32
+
+    When working with statistical hypothesis testing, we usually use a
+    threshold probability or significance level upon which we decide
+    to reject the null hypothesis :math:`H_0`. Suppose we choose the common
+    significance level of 5%.
+
+    Our alternative hypothesis is that College Professors are trully more
+    satisfied in their works than Scientists. Therefore, we expect
+    :math:`p_1` the proportion of very satisfied College Professors to be
+    greater than :math:`p_2`, the proportion of very satisfied Scientists.
+    We thus call `boschloo_exact` with the ``alternative="greater"`` option:
+
+    >>> import scipy.stats as stats
+    >>> res = stats.boschloo_exact([[74, 31], [43, 32]], alternative="greater")
+    >>> res.statistic
+    0.0483...
+    >>> res.pvalue
+    0.0355...
+
+    Under the null hypothesis that Scientists and College Professors are
+    equally satisfied of their works, the probability of obtaining test
```

Comment:
Which is the null hypothesis? That they are "equally satisfied" or that "H_0 : p_1 \leq p_2" as is written above? I suspect the null distribution comes from the former and that the `alternative` keyword does not change the null hypothesis - only the alternative.

Reply:
It is a mistake from my side and I am going to change it :-)